src/EditorFeatures/CSharpTest/Diagnostics/Suppression/RemoveUnnecessaryPragmaSuppressionsTests.cs

```diff
@@ -265,23 +265,34 @@ void M()
                 }
             }
 
+            public enum TestKind
+            {
+                Pragmas,
+                SuppressMessageAttributes,
+                PragmasAndSuppressMessageAttributes
+            }
+
             [Theory, CombinatorialData]
-            public async Task TestDoNotRemoveUnsupportedDiagnosticSuppression(bool disable)
+            [WorkItem(46047, "https://github.com/dotnet/roslyn/issues/46047")]
+            public async Task TestDoNotRemoveUnsupportedDiagnosticSuppression(bool disable, TestKind testKind)
```

Comment:
The exception was being hit for `TestKind.SuppressMessageAttributes` when there are no pragmas in input.